src/main/webapp/app/core/login/login.service.ts

```diff
@@ -59,14 +61,19 @@ export class LoginService {
      * Tokens, Alerts, User object in memory.
      * Will redirect to home when done.
      */
-    logout() {
+    logout(wasInitiatedByUser: boolean) {
+        this.logoutWasForceful = !wasInitiatedByUser;
+
         this.authServerProvider
             // 1: Clear the auth tokens from the browser's caches.
             .removeAuthTokenFromCaches()
             .pipe(
                 // 2: Clear all other caches (this is important so if a new user logs in, no old values are available
                 tap(() => {
-                    return this.authServerProvider.clearCaches();
+                    if (wasInitiatedByUser) {
```

Comment:
this might become a problem, if a Student logs out on purpose and a another Student logs himself in. 
For example this might occur very often during testing sessions. Maybe we can find a different way which is specific for exams?

Reply:
Good observation; that is why we clear the caches when a student logs out on purpose. @krusche already proposed I implement a wrapper service in a follow-up PR which makes sure we only load data from the respective login.